go/ngql/types.go

```diff
@@ -419,32 +419,41 @@ func getListElements(vrw types.ValueReadWriter, v types.Value, args map[string]i
 	}
 
 	// Clamp ranges
-	if count <= 0 || idx >= len {
+	if count <= 0 || idx >= end {
 		return ([]interface{})(nil)
 	}
 	if idx < 0 {
 		idx = 0
 	}
-	if idx+count > len {
-		count = len - idx
+	if idx+count > end {
+		count = end - idx
 	}
 
 	values := make([]interface{}, count)
 
-	// This is a workaround for allowing types.Set as types.List.
-	type listOrSetIterator interface {
-		Next() types.Value
-	}
+	// @raf a few questions:
+	//  - why do we do MaybeGetScalar below?
+	//  - in the unittests how could you force the collection to have tree height > 1?
+	//  - where does the rejiggering of the tree height happen?
+	//  - i just ran the unit tests; what else should i do to make sure this change is WAI?
```

Comment:
There are some tests that keep track of number of reads and writes to a chunk store. Maybe you could keep track of the number of reads to see that we read all the chunks even if exit early (can we exit early?)

Checking that we do not read sequentially seems flaky since even with random order sequential order is possible so that does not seem viable...

Reply:
Oh! You could use a ValueReadWriter that logs the number of calls to `ReadManyValues`. That used to be 0 in this scenario but now should be 1.